docs/madr/decisions/050-meshexternalservice-api.md

```diff
@@ -0,0 +1,679 @@
+      port: 12345
+    - address: example.com
+    - address: unix://....
+  tls:
+    version:
+      min: TLS12 # or TLSAuto, TLS10, TLS11, TLS12, TLS13
+      max: TLS13 # or TLSAuto, TLS10, TLS11, TLS12, TLS13
+    allowRenegotiation: false
+    verification:
+      enabled: true
+      skipVerification: Secured
+      subjectAltNames: # if subjectAltNames is not defined then take domain or ips
+        - type: Exact
+          value: example.com
+        - type: Prefix
+          value: "spiffe://example.local/ns/local"
+      caCert: 
+        inline: 123
+      clientCert:
+        secret: 123
+      clientKey:
+        secret: 123
+status: 
+  vip:
+    value: 242.0.0.1
+    type: Kuma
+  addresses:
+  - hostname: example.ext.svc.local
+    status: NotAvailable
+    origin:
+      kind: HostnameGenerator
+      name: k8s-example-service-hostname
+    reason: "addresses are overlapping with ext2"
+```
+* **spec**:
+  * **match**: defines traffic that should be routed through the sidecar
+    * **type**: type of the match, only `HostnameGenerator` is available at the moment
+      * `HostnameGenerator`: matches hostnames provided by HostnameGenerator
+    * **port**: defines a port to which a user does requests
+    * **protocol**: defines a protocol of the communication. Possible values:
+      * `tcp`: WARNING: shouldn't be used when match is `Domain` type. On the TCP level we are not able to disinguish domain, in this case it is going to hijack whole traffic on this port. We are going to validate configuration and do not apply config when protocol is tcp and `type: Domain`.
+      * `grpc`
+      * `http`
+      * `http2`
+  * **extension**: struct for a plugin configuration, in the presence of an extension `endpoints` and `tls` are not required anymore - it's up to the extension to validate them independently.
+    * **type**: defines what kind of plugin to use, it's a string type so any new plugins should work.
+    * **config**: json map that is mapped to configuration provided in the type.
+  * **endpoints**: defines a list of endpoints.
+    * **address**: defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets
+    * **port**: defines a port of a destination.
+  * **tls**: provides a TLS configuration when proxy is resposible for a TLS origination
+    * **enabled**: defines if proxy should originate TLS.
+    * **version**: section for providing version specification.
+      * **min**: defines minmum supported version. One of `TLSAuto`, `TLS10`, `TLS11`, `TLS12`, `TLS13`
+      * **max**: defines maximum supported version. One of `TLSAuto`, `TLS10`, `TLS11`, `TLS12`, `TLS13`
+    * **allowRenegotiation**: defines if TLS sessions will allow renegotiation.
+    * **verification**: section for providing TLS verification details.
+      * **skipVerification**: defines if proxy should skip verification, one of `SkipSAN`, `SkipCA`, `Secured`, `SkipALL`. Default `Secured`.
```

Comment:
I think based on the comment it should be `mode`, otherwise it looks weird

```suggestion
      * **mode**: defines if proxy should skip verification, one of `SkipSAN`, `SkipCA`, `Secured`, `SkipALL`. Default `Secured`.
```